inc/Integrations/Airtable/AirtableIntegration.php

```diff
@@ -7,6 +7,10 @@
 
 class AirtableIntegration {
 	public static function init(): void {
+		add_action( 'init', [ __CLASS__, 'register_blocks' ] );
```

Comment:
This moves block registration to the `init` action rather than in `::init()`. The latter is called as the plugin files are loaded, but not necessarily all of WordPress. This doesn't fix any particular issue right now, but in general it's good to wait until the WordPress `init` action has fired before doing any data manipulations, as all of WordPress (and all plugin files) may not have been loaded yet.

Also changed below in `ShopifyIntegration`.

Reply:
I think we should move this out to `remote-data-blocks.php` and call `AirtableIntegration::init()` from inside a callback there instead of in the `init` function here. I think once this function is called you should be able to expect the blocks to be registered.

Reply:
I'm happy to change this if you still disagree, but this is the general pattern we use for almost every `::init()` from `remote-data-blocks.php`:

https://github.com/Automattic/remote-data-blocks/blob/b49bf7b31d00ccab8ef6aaf07f59cdb00477687e/inc/Editor/AdminNotices/AdminNotices.php#L32-L35

https://github.com/Automattic/remote-data-blocks/blob/b49bf7b31d00ccab8ef6aaf07f59cdb00477687e/inc/Editor/DataBinding/BlockBindings.php#L17-L20

https://github.com/Automattic/remote-data-blocks/blob/b49bf7b31d00ccab8ef6aaf07f59cdb00477687e/inc/Editor/DataBinding/FieldShortcode.php#L8-L10

https://github.com/Automattic/remote-data-blocks/blob/b49bf7b31d00ccab8ef6aaf07f59cdb00477687e/inc/Editor/DataBinding/QueryOverrides.php#L10-L12

https://github.com/Automattic/remote-data-blocks/blob/b49bf7b31d00ccab8ef6aaf07f59cdb00477687e/inc/Editor/BlockManagement/BlockRegistration.php#L23-L26

https://github.com/Automattic/remote-data-blocks/blob/b49bf7b31d00ccab8ef6aaf07f59cdb00477687e/inc/Editor/PatternEditor/PatternEditor.php#L10-L13

https://github.com/Automattic/remote-data-blocks/blob/b49bf7b31d00ccab8ef6aaf07f59cdb00477687e/inc/ExampleApi/ExampleApi.php#L14-L16

https://github.com/Automattic/remote-data-blocks/blob/b49bf7b31d00ccab8ef6aaf07f59cdb00477687e/inc/PluginSettings/PluginSettings.php#L15-L21

https://github.com/Automattic/remote-data-blocks/blob/b49bf7b31d00ccab8ef6aaf07f59cdb00477687e/inc/REST/RemoteDataController.php#L16-L18

---

The only classes we don't hook into the WordPress lifecycle are in our logging and Tracks setup:

https://github.com/Automattic/remote-data-blocks/blob/b49bf7b31d00ccab8ef6aaf07f59cdb00477687e/inc/Editor/BlockManagement/ConfigRegistry.php#L21-L24

https://github.com/Automattic/remote-data-blocks/blob/b49bf7b31d00ccab8ef6aaf07f59cdb00477687e/inc/Analytics/TracksAnalytics.php#L28-L46

I could see why we may want these working prior to WordPress startup for metrics and error logging. However, even the logger internally calls this during constructor during `::init()`:

https://github.com/Automattic/remote-data-blocks/blob/b49bf7b31d00ccab8ef6aaf07f59cdb00477687e/inc/Logging/Logger.php#L65

This is an anti-pattern because only code that's loaded into memory before our plugin can register to use the `wpcomvip_log_level` hook. Anything loaded after won't be able to modify this hook, because it's already been called. This may not matter for this particular filter, but this is a potential footgun on all code interacting with WordPress features before `init`.

In summary, most of our classes already wait until the `init` hook (or other lifecycle actions) to initialize class data internally. We have the option to initialize classes before that, but it's easy to include mistakes that don't work well with other WordPress code. Unless we really need global access to a class prior to `init` and are careful about which WordPress features we use, we'll eventually run into file load-order, uninitialized global state, and other weird initialization bugs. I think the best bet is to rely on lifecycle actions and priority for this rather than carefully coding around load-order bugs unless we have a great reason.